cirq/sim/clifford/clifford_simulator.py

```diff
@@ -366,14 +379,26 @@ def apply_single_qubit_unitary(self, op: 'cirq.Operation'):
         phase_shift = u[max_idx] / applied_unitary[max_idx]
         self.ch_form.omega *= phase_shift
 
-    def _apply_H(self, qubit: ops.Qid):
+    def _apply_H(self, qubit: int):
         self.tableau._H(qubit)
         self.ch_form._H(qubit)
 
-    def _apply_S(self, qubit: ops.Qid):
+    def _apply_S(self, qubit: int):
         self.tableau._S(qubit)
         self.ch_form._S(qubit)
 
+    def _apply_X(self, qubit: int):
+        self.tableau._X(qubit)
+        self.ch_form._X(qubit)
+
+    def _apply_Z(self, qubit: int):
+        self.tableau._Z(qubit)
+        self.ch_form._Z(qubit)
+
+    def _apply_Y(self, qubit: int):
+        self.tableau._Y(qubit)
+        self.ch_form._Y(qubit)
+
```

Comment:
Also add and use _apply_CZ and _apply_CNOT for completeness.

Reply:
@viathor Nothing really. Ideally we can take this to conclusion by removing the _GATE methods on the tableau and CH form too and let individual gates modify the internal states of tableau and the CH form. It will be more involved but I can make that change.

Do we want to have two separate protocols: `apply_clifford_tableau` and `apply_stabilizer_ch_form` or combine them into one `update_clifford_state`? (names subject to change)

@mpharrigan Feel free to ignore the initial suggestion in this comment.